src/qml/pages/NewChatPage.qml

```diff
@@ -9,64 +10,288 @@ import "../theme"
 OperationPage {
     appState: app.stateManager.newChatState
     loadingText: qsTr("Adding of contact...")
-    readonly property string contact: username.text.toLowerCase()
+    footerText: ""
+    readonly property string contact: contactSearch.search.toLowerCase()
+    property var filterSource: models.discoveredContacts
+    property alias search: contactSearch.search
+    property int modelCount: contactListView.count
+    property string searchResultState: "search empty"
+    property string serverName: "Default"
+
+    readonly property int defaultMargins: 20
+    readonly property int defaultSearchHeight: 40
+    readonly property int defaultChatHeight: 50
+
+    onSearchChanged: {
+        if (filterSource) {
+            filterSource.filter = search
+            checkState()
+        }
+    }
+
+    onModelCountChanged: {
+        checkState()
+    }
 
     header: Header {
-        showBackButton: !form.isLoading
+//        showBackButton: !form.isLoading
+        showBackButton: true
         title: qsTr("New chat")
     }
 
-    Form {
-        id: form
-        formSpacing: 25
-        focus: true
+    Search {
+        id: contactSearch
+        anchors {
+            top: parent.top
+            left: parent.left
+            right: parent.right
+            margins: defaultMargins
+        }
+        height: defaultSearchHeight
+        state: "opened"
+        searchPlaceholder: qsTr("Search contact")
+        onClosed: {
+            contactSearch.search = ""
+            reject()
+        }
+        onAccepted: {
+            accept()
+        }
+    }
 
-        UserNameFormInput {
-            id: username
-            label: qsTr("Username")
-            focus: true
+    Item {
+        id: searchResultsItem
+        property real headerHeight: 20
+        property real headerOpacity: 0
+
+        anchors {
+            top: contactSearch.bottom
+            left: parent.left
+            right: parent.right
+            bottom: parent.bottom
+            topMargin: 1
+            bottomMargin: defaultChatHeight
+            margins: defaultMargins
+        }
+
+        state: searchResultState
+        states: [
+            State {
+                name: "not found" // search word is NOT IN model
```

Comment:
Comments are obvious.
I prefer to use identifiers in name, for example: notFound. You are free to prefer too)

Reply:
Nooo